### PR TITLE
w3c tracing integration test

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Added integration test for W3C tracing

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #394 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: sdk-tests/src/test/java/io/dapr/it/tracing/http/OpenTelemetryInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    Context context = textFormat.extract(Context.current(), request, HTTP_SERVLET_REQUEST_GETTER);
+    request.setAttribute("opentelemetry-context", context);
+    return true;
+  }
+
+  @Override
+  public void postHandle(
+      HttpServletRequest request, HttpServletResponse response, Object handler,
+      ModelAndView modelAndView) {
+  }
```
Comment (halspang) [inline sdk-tests/src/test/java/io/dapr/it/tracing/http/OpenTelemetryInterceptor.java:61]: Can we have a comment in this stating that it's intentionally left empty?

**Review by halspang — COMMENTED** [review]
Looks fine overall, just a few quick comments.

File: sdk-tests/src/test/java/io/dapr/it/tracing/OpenTelemetry.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    return reactorContext;
+  }
+
+  private static boolean isZipkinUp() {
+    try (Socket ignored = new Socket("localhost", ZIPKIN_PORT)) {
+      return true;
+    } catch (IOException ignored) {
+      return false;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (artursouza) [reply sdk-tests/src/test/java/io/dapr/it/tracing/OpenTelemetry.java:99]: If that happens, then we need to have a smarter logic here. Zipkin is installed way before this check, so we don't run into that race condition (yet).

File: sdk-tests/src/test/java/io/dapr/it/tracing/Validation.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import okhttp3.Request;
+import okhttp3.Response;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Class used to verify that traces are present as expected.
+ */
+public final class Validation {
```
[the remark this answers is not in the document]
Comment (artursouza) [reply sdk-tests/src/test/java/io/dapr/it/tracing/Validation.java:22]: Added more comments.

File: sdk-tests/src/test/java/io/dapr/it/tracing/Validation.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    // Must wait for some time to make sure Zipkin receives all spans.
+    Thread.sleep(5000);
+    HttpUrl.Builder urlBuilder = new HttpUrl.Builder();
+    urlBuilder.scheme("http")
+        .host("localhost")
+        .port(9411);
+    urlBuilder.addPathSegments("api/v2/traces");
+    Request.Builder requestBuilder = new Request.Builder()
+        .url(urlBuilder.build());
+    requestBuilder.method("GET", null);
```
[the remark this answers is not in the document]
Comment (artursouza) [reply sdk-tests/src/test/java/io/dapr/it/tracing/Validation.java:41]: addPathSegments, yes. Method, no. Done.

File: sdk-tests/src/test/java/io/dapr/it/tracing/http/OpenTelemetryInterceptor.java (hunk 1 above)
Comment (artursouza) [reply sdk-tests/src/test/java/io/dapr/it/tracing/http/OpenTelemetryInterceptor.java:61]: Done.

**Review by mukundansundar — APPROVED** [review]

**Review by halspang — APPROVED** [review]

**Pull request merged**